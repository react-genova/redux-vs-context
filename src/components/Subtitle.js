import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Subtitle = ({ subtitle, className }) => {
    console.log("render Subtitle");
    return <h2 className={className}>{subtitle}</h2>;
};

Subtitle.propTypes = {
    className: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};

const SubtitleStyled = styled(Subtitle)`
    margin: 0 10px;
    text-align: left;
    text-transform: capitalize;
    color: #8d8;
`;

export default SubtitleStyled;
