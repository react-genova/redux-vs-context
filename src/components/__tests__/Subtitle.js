import React from 'react';
import Subtitle from '../Subtitle';
import { render } from 'react-testing-library';
import 'jest-styled-components';

describe('Subtitle component', () => {
    global.console = { log: () => null };

    it('renders without crashing', () => {
        const { container } = render(
            <Subtitle subtitle="the subtitle" className="theclassname" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
