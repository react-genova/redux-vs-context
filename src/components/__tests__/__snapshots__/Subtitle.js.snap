// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle component renders without crashing 1`] = `
.c0 {
  margin: 0 10px;
  text-align: left;
  text-transform: capitalize;
  color: #8d8;
}

<h2
  class="theclassname c0"
>
  the subtitle
</h2>
`;
